Week-2 Updated code - To be reviewed

amanrajs opened the pull request.

29amansingh committed:
Week-2 Updated code - To be reviewed

29amansingh committed:
WEEK 2 TASKS CHANGES DONE , used constants and followed coding practise

mnhaque commented on Week-2/style/style.css:
```diff
@@ -0,0 +1,170 @@
+  padding:10px;
+}
+.closeButton{
+}
```
we can remove this

mnhaque commented on Week-2/scripts/RenderContent.js:
```diff
@@ -0,0 +1,76 @@
+  let data=[];
+  for(item in article_data){
+    if(data.indexOf(article_data[item].postCategory)!==-1){
+      continue;
```
generally `if else` block is indented as below example

```
if(condition) {
  //code
} else if(condition) {
  //code
} else {
  //code
}
```

mnhaque commented on Week-2/scripts/RenderContent.js:
```diff
@@ -0,0 +1,76 @@
+  fullContainer.appendChild(right_container);
+  select.options[0].disabled = true;
+  select.onchange = () => {
+    let val=select.value;
```
can get rid of this line and use the expression in below line.

mnhaque commented on Week-2/scripts/RenderContent.js:
```diff
@@ -0,0 +1,76 @@
+    let val=select.value;
+    changeContent(val,container);
+  }
+}
```
consider splitting this function to few smaller functions

mnhaque commented on Week-2/scripts/constants.js:
```diff
@@ -0,0 +1,89 @@
+const defaultContent='Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proi ...
+const defaultDescription='created at 21 July 2019""created at 21 July 2019';
```
can use single const declaration for all variables.

mnhaque commented on Week-2/scripts/RenderContent.js:
```diff
@@ -0,0 +1,76 @@
+    card.appendChild(right);
+    container.appendChild(card);
+    container.appendChild(horizontalline);
+    fullContainer.appendChild(container);
```
consider splitting this function into few smaller functions

mnhaque commented on Week-2/scripts/utility.js:
```diff
@@ -0,0 +1,41 @@
+  if(regexEmail.test(emailid)){
+    window.localStorage.setItem(emailid, JSON.stringify(emailid));
+    alert(emailSubscribedMessage);
+  }
```
generally `if else` block is indented as below example

```
if(condition) {
  //code
} else if(condition) {
  //code
} else {
  //code
}
```

mnhaque commented on Week-2/scripts/utility.js:
```diff
@@ -0,0 +1,41 @@
+      popup.appendChild(descriptionPopup);
+      fullContainer.style.opacity = '0.3';
+    }
+    else if(clickEvent.srcElement.id === 'closePopup'){
```
generally `if else` block is indented as below example

```
if(condition) {
  //code
} else if(condition) {
  //code
} else {
  //code
}
```

mnhaque commented on Week-2/scripts/RenderContent.js:
```diff
@@ -0,0 +1,76 @@
+  right_container.appendChild(titleSelectCategory);
+  right_container.appendChild(select);
+  right_container.appendChild(subscribeTitle);
+  right_container.appendChild(check);
```
few places same dom manipulation function used, this can be clubbed by making a few common functions. this can reduce a lot of code lines.
discuss with me if you are not clear what i am telling

29amansingh committed:
did the changes reviewed
-> using a common createCard function for both static and news data,to reduce dom code in a single function
->other minor changes

amanrajs committed:
Update RenderContent.js

mnhaque commented:
looks good
